
<!-- saved from url=(0072)http://amp.pharm.mssm.edu/Enrichr/templates/help/background-submenu.html -->
<html class="gr__amp_pharm_mssm_edu"><head><meta http-equiv="Content-Type" content="text/html; charset=windows-1252"><style type="text/css" abt="234"></style></head><body data-gr-c-s-loaded="true"><ul>
	<li>What are regulons and CTS-Rs and why are they important?</li>
	<li>Why allow us to store your data?</li>
	<li>What is the hypergeometric test?</li>
	<li>What are the evaluation indexes for predicted cell types?</li>
	<li>How do DMINDA and MEME predict regulons?</li>
	<li>Why not using the TF name to represent the regulon or motif name?</li>
	<li>How does CeRIS define and interpret regulon names?</li>
	<li>How does CeRIS rank regulons?</li>
	<li>How does CeRIS perform comparison of CTS-Rs to ATAC-Seq peaks?</li>
	<li>How does CeRIS use TAD to retrive additional genes?</li>
	<li>How does CeRIS perform normalization for CTS-R heatmaps?</li>
	<li>Why is my heatmap not showing up?</li>
	<li>Why am I getting 404 error when open the Enrichr, JASPAR and HOCOMOCO links?</li>
	<li>How is the performance of CeRIS?</li>
	<li>What are included in the Download zip file?</li>
	<li>Why I'm staying in the confirmation page when the job has been done?</li>
	<li>How do I report errors?</li>
	<li>Where can I find the source code?</li>
	<li>What are the future plans of CeRIS?</li>
	<li class="last">Brower compatibility of CeRIS.</li>
</ul></body></html>