
<!-- saved from url=(0068)http://amp.pharm.mssm.edu/Enrichr/templates/help/basics-content.html -->
<html class="gr__amp_pharm_mssm_edu"><head><meta http-equiv="Content-Type" content="text/html; charset=windows-1252"><style type="text/css" abt="234"></style></head><body data-gr-c-s-loaded="true"><div id="the-basics-content">
	<h1>FAQ</h1>
	<div class="qa">
		<h3 class="question"><a id="Q0" style="color:black;">What are regulons and CTS-Rs and why are they important?</a></h3>
		<div class="answer">
			<p>The term regulon was first introduced by Maas et al. in 1964, intended to name a maximal group of co-regulated genes, 
			which may be scattered in a genome without apparent patterns in terms of their genomic locations. Based on that, for the 
			first time, we define a CTS-R, cell-type-specific regulon, as a group of genes, receiving similar regulatory signals in a specific cell type, 
			hence tend to have similar expression patterns and share conserved motifs in this cell type. A successful elucidation of 
			CTS-Rs will substantially improve the identification of transcriptionally co-regulated gene modules, realistically 
			allowing reliable prediction of global transcription regulation networks encoded in a specific cell type. </p>
			<br>
		</div>
		<div class="clear"></div>
	</div>
	
	<div class="qa">
		<h3 class="question"><a id="Q1" style="color:black;">Why allow us to store your data?</a></h3>
		<div class="answer">
			<p>Our long-term goal is to build a cell-type-specific regulon atlas for diverse cell types across different species 
			to uncover the heterogeneous regulatory mechanisms in different cell types. Your data could potentially help us set 
			up the database, and all contributions will be acknowledged in our future work.</p>
			<br>
		</div>
		<div class="clear"></div>
	</div>
	
	<div class="qa">
		<h3 class="question"><a id="Q2" style="color:black;">What is the hypergeometric test?</a></h3>
		<div class="answer">
			<p>The hypergeometric enrichment test is performed in both CTS-gene module assignment and the functional enrichment for CTS-R evaluations. To infer CTS-gene modules, the probability of having x cells of the same cell type in a bicluster of size n from the dataset with a total of N cells can be computed using the following hypergeometric function:</p>
			<br> <a href="./assets/img/tutorial/4.Q2.png"></a>
			<p>where p is the percentage of that cell type among all cell types in the data set. The p-value of getting such enriched bicluster is calculated as:</p>
			<br> <a href="./assets/img/tutorial/4.Q2.2.png"></a>
			<p>The p-value of a bicluster corresponding to a specific cell type is Bonferroni-adjusted by multiplying by the total number of cell types, and such bicluster is assigned as a CTS-gene module if its adjusted p-value to that cell type is significant (adj.p<0.05). </p>
			<p>The functional enrichment analysis between genes in the CTS-R and databases, such as KEGG, GO, ProteomicDB, etc. uses the same hypergeometric test and p-value adjustment shown above, but compares genes instead of cells. All the enrichment analysis in IRIS3 are performed using Enrichr. </p>
			<br>
		</div>
		<div class="clear"></div>
	</div>

	<div class="qa">
		<h3 class="question"><a id="Q3" style="color:black;">What are the evaluation indexes for predicted cell types?</a></h3>
		<div class="answer">
			<p>Four index scores, Rand Index (RI), Adjusted Rand Index (ARI), Fowlkes and Mallows's index (FMI), and Jaccard Index (JI) are calculated to evaluate the similarity between the prediction and ground-truth labels. A Silhouette score showing how similar each cell is to its type compared to other clusters is also calculated. If the ground-truth cell type file is not provided by users, both the index scores and Silhouette score are omitted since no comparing target can be accessed. </p>
			<br>
		</div>
		<div class="clear"></div>
	</div>
	
	<div class="qa">
		<h3 class="question"><a id="Q4" style="color:black;">How do DMINDA and MEME predict regulons?</a></h3>
		<div class="answer">
			<p>DMINDA: is an integrated web server for DNA motif 
			identification and analyses. The server provides a suite of cis-regulatory motif analysis functions on DNA 
			sequences, which are important to elucidation of the mechanisms of transcriptional regulation: (i) de novo 
			motif finding for a given set of promoter sequences along with statistical scores for the predicted motifs 
			derived based on information extracted from a control set, (ii) scanning motif instances of a query motif in 
			provided genomic sequences, (iii) motif comparison and clustering of identified motifs, and (iv) co-occurrence 
			analyses of query motifs in given promoter sequences.</p>
			<br>
			<p>MEME suite: supports motif-based analysis of DNA, RNA and protein sequences. 
			It provides motif discovery algorithms using both probabilistic and discrete models, which have complementary 
			strengths. It also allows the discovery of motifs with arbitrary insertions and deletions (GLAM2). In addition 
			to motif discovery, the MEME Suite provides tools for scanning sequences for matches to motifs (FIMO, MAST 
			and GLAM2Scan), scanning for clusters of motifs (MCAST), comparing motifs to known motifs (Tomtom), finding 
			preferred spacings between motifs (SpaMo), predicting the biological roles of motifs (GOMo), measuring the 
			positional enrichment of sequences for known motifs (CentriMo), and analyzing ChIP-seq and other large datasets (MEME-ChIP).</p>
			<br>
		</div>
		<div class="clear"></div>
	</div>
	
	<div class="qa">
		<h3 class="question"><a id="Q5" style="color:black;">Why not using the TF name to represent the regulon or motif name?</a></h3>
		<div class="answer">

			<p>The query motif identified by IRIS3 or MEME may match to more than one target TFBS profiles in the 
			motif databases from TOMTOM (see the following figure). Hence, it is not feasible to name the predicted 
			motif or regulon by one TF name.</p>
			<p>Therefore, the motif may not be unique, and two or more share the similarity may exist in one CTS-R.</p>
			<img src="./assets/img/tutorial/4.Q5.png">
			<br>
		</div>
		<div class="clear"></div>
	</div>
	
	<div class="qa">
		<h3 class="question"><a id="Q6" style="color:black;">How does IRIS3 define and interpret regulon names?</a></h3>
		<div class="answer">
			<p>For each CTS-R, its name is defined by the cell type number and the assigned regulons. For example: </p>
			<img src="./assets/img/tutorial/4.Q6.png">
			<br>
		</div>
		<div class="clear"></div>
	</div>
	
	<div class="qa">
		<h3 class="question"><a id="Q7" style="color:black;">How does IRIS3 rank regulons?</a></h3>
		<div class="answer">
			<p>Each regulon is a set of genes that correspond to the predicted motif based on the motif scanning and 
			prediction (see <a href="#Q4"><b>FAQ above</b></a>). A group of motifs predicted by DMINDA is then be clustered into subgroups composed 
			of similar motifs using the following algorithm, which is based on a maximum spanning tree (MST) representation 
			of the candidate motifs. In this way, it is possible that more than one motif with similarities are assigned to 
			one regulon and the genes are then the union of all motifs. A global P-value for the entire motif instances is 
			calculated during motif scanning by summing up the possibility of at least k instances denoted in the motif. The 
			most significant p-value is used as the CTS-R p-value. The system will then sort and name regulons based 
			on the p-values from low to high.</p>
			<br>
		</div>
		<div class="clear"></div>
	</div>
	
	<div class="qa">
		<h3 class="question"><a id="Q18" style="color:black;">How does IRIS3 perform comparison of CTS-Rs to ATAC-Seq peaks?</a></h3>
		<div class="answer">
			<p>Genes can be regulated only if their promoter regions are located in the open chromatin area that is available for the regulator to bind. We downloaded several ATAC-Seq data of human (74 tissue) and mouse (117) from <a href="http://cistrome.org/db/#/"><b>CistromeDB</a></b>. Specifically, we compare all motifs in a CTS-R to the peak regions from the ATAC-Seq. If a motif instance overlaps with the peak region, it will be considered as an open chromatin region enriched motif which is available for the binding of the corresponding TF. Genes in the CTS-R and supported by the ATAC-Seq are then validated as a true CTS-gene. However, such chromatin activities (peaks) are diverse in different cell types, or even cells. The bulk ATAC-Seq data we found for different tissues/cell types cannot provide accurate validation to CTS-genes. Thus, such function is only additionally provided to users to have a general glance. In the future, we are expecting to have single-cell ATAC-Seq data pairing with scRNA-Seq data for credible gene validations in CTS-Rs.  </p>
			<br>
		</div>
		<div class="clear"></div>
	</div>	
	
	<div class="qa">
		<h3 class="question"><a id="Q19" style="color:black;">How does IRIS3 use TAD to retrive additional genes?</a></h3>
		<div class="answer">
			<p>Considering the computational efficiency, the region for motif finding in IRIS3 is limited to 1000-2000 bp region upstream to each gene, and motifs outside of the region may not be found. To amend such a deficiency, we integrated <b>TAD (Topologically Associated Domain)</b> data resulted from Hi-C to trace back genes in the CTS-gene modules of a cell type but not in the CTS-R. Genes in the same TAD are more likely to be regulated together due to the topological folding structure of such region. We first downloaded 19 human tissue and five mouse tissues TAD files, respectively, and identified all genes for each TAD region per file. TAD regions are selected if any genes in a CTS-R are found in the TAD gene list. The remaining TAD-genes are used to search in the CTS-gene modules of that CTS-R and output all matched genes as TAD supplementary genes.</p>
			<br>
		</div>
		<div class="clear"></div>
	</div>	

	<div class="qa">
		<h3 class="question"><a id="Q8" style="color:black;">How does IRIS3 perform normalization for CTS-R heatmaps?</a></h3>
		<div class="answer">
			<p>To better illustrate the expression value of the gene sets, we applied a <a href="https://f1000research.com/articles/5-2122/v1"><b>log-normalization</b></a> for heatmap interpretation:</p>
			<img src="./assets/img/tutorial/4.Q8-1.png">
			<br>
			<br>
			<br>
			<p>Where, <b><i>VN<sub>i</sub></i></b> indicates the normalized value for gene <b><i>i</i></b>, <b><i>VE<sub>i</sub></i></b> 
			indicates the expression value of gene <b><i>i</i></b>, and <b><i>n</i></b> is the number of total cells. The heatmap 
			comparison of pre-normalized and normalized values are shown below.</p>
			<img src="./assets/img/tutorial/4.Q8-2.png">
			<br>
		</div>
		<div class="clear"></div>
	</div>

	<div class="qa">
		<h3 class="question"><a id="Q9" style="color:black;">Why is my heatmap not showing up?</a></h3>
		<div class="answer">
			<p>The Clustergrammer is a real-time computing tool that generates the interactive heatmaps after open 
			the webpage. Based on the complexity of the matrix (total number of genes in the CTS-Rs and the 
			total number of cells of your data), the time various (normally less than 10 seconds). Please be patient 
			while waiting for the displaying.</p>
			<br>
		</div>
		<div class="clear"></div>
	</div>

	<div class="qa">
		<h3 class="question"><a id="Q10" style="color:black;">Why am I getting 404 error when open the Enrichr, JASPAR and HOCOMOCO links?</a></h3>
		<div class="answer">
			<p>All these three functions use the outside server. Please ensure your browser allows the pop-up windows. 
			Sometimes, an 404 error page shows up due to the temperal server loading error and once you refresh the error page or click the 
			link button again, the result page can showup successfully</p>
			<br>
		</div>
		<div class="clear"></div>
	</div>
	
	<div class="qa">
		<h3 class="question"><a id="Q11" style="color:black;">How is the performance of IRIS3?</a></h3>
		<div class="answer">
			<p>To evaluate the performance of IRIS3, we compared our regulons with the result predicted by the 
			<a href="https://github.com/aertslab/SCENIC"><b>SCENIC pipeline</b></a>, using the example data. 
			Default parameters were used for both tools.</p>
			<img src="./assets/img/tutorial/4.Q10-1.png">
			<p>As a result, our test showed that: 
			69.60% (316/454) and 39.74% (184/463) of regulons identified by IRIS3 and SCENIC were significantly enriched with at least one KEGG pathway among all 293 pathways. IRIS3 achieved an F1-score of 0.673 compared to the 0.547 in SCENIC, indicating a better performance of IRIS3 related to the prediction of more significant CTS-Rs. In addition, IRIS3 only spent 42 minutes to complete the prediction, while SCENIC spent 9.5 hours under the same computational configuration. 
			<p><b>Note:</b> The harmonic mean of precision and recall were calculated to represent the performance 
			of an algorithm on a given dataset and parameter setting, denoted as F-score:</p>
			<img src="./assets/img/tutorial/4.Q10-2.png">
			<img src="./assets/img/tutorial/4.Q10-3.png">
			<img src="./assets/img/tutorial/4.Q10-4.png">
			<br>
		</div>
		<div class="clear"></div>
	</div>	

	<div class="qa">
		<h3 class="question"><a id="Q12" style="color:black;">What are included in the Download zip file?</a></h3>
		<div class="answer">
			<img src="./assets/img/tutorial/4.Q12.png">
			<p>1. The conds file contains cell names and the gene file contains gene names extracted from each bicluster (file 5), respectively. Each row represent a bicluster. </p>
			<p>2. The cell type prediction result from SC3</p>
			<p>3. The gene list for each CTS-R, grouped by cell types. Each row indicates a regulon.</p>
			<p>4. The filtered gene expression matrix. Omitted if user chooses not to perform the gene filtering step.</p>
			<p>5. The biclustering result from QUBIC.</p>
			
			<br>
		</div>
		<div class="clear"></div>
	</div>	

	<div class="qa">
		<h3 class="question"><a id="Q13" style="color:black;">Why I'm staying in the confirmation page when the job has been done?</a></h3>
		<div class="answer">
			<p>The confirmation page and the result page share the same link address. When click the result link in the email, some users may still be 
			navigated to the confirmation page due to the cache stored in the browser. Just refresh the page or wait 15 sec for automatic refresh, and 
			your result will show up.</p>
			<br>
		</div>
		<div class="clear"></div>
	</div>	
	
	<div class="qa">
		<h3 class="question"><a id="Q14" style="color:black;">How do I report errors?</a></h3>
		<div class="answer">
			<p>Please report any errors or comments on the <a href="https://groups.google.com/forum/#!forum/iris3-discussion-board"><b>
			IRIS3 google group</b></a>, or directly email us at <b>ma.1915@osu.edu</b>.</p>
			<br>
		</div>
		<div class="clear"></div>
	</div>	

	<div class="qa">
		<h3 class="question"><a id="Q15" style="color:black;">Where can I find the source code?</a></h3>
		<div class="answer">
			<p>The source code of IRIS3 can be accessed on GitHub at <b>https://github.com/OSU-BMBL/IRIS3</b></p>
			<br>
		</div>
		<div class="clear"></div>
	</div>	

	<div class="qa">
		<h3 class="question"><a id="Q16" style="color:black;">What are the future plans of IRIS3?</a></h3>
		<div class="answer">
			<ul>
			<li>Build up a database of CTS-R atlas in diverse cell types of human, mouse and extend to other species.</li>
			<br>
			<li>Integrate more sequence analysis, including quality control, alignment, mapping, and normalization in the 
			IRIS3 pipeline to generate the normalized expression data from FASTQ.</li>
			<br>
			<li>An in-development tool, BRIC (biclustering-based gene regulation and cell type prediction from single-cell 
			RNA-Seq data), will be used to replace QUBIC to predict cell types and CTS-Rs simultaneously. The BRIC is 
			integrated with (1) a left truncated mixture Gaussian model for the accurate quantification of the sample-specific 
			gene regulatory signals and (2) a dual-core algorithm for more precise biclusters identification.</li>
			<br>
			<li>A more powerful in-house deep learning pipeline, DESSO, will be added in IRIS3 as a replacement of DMINDA and 
			MEME to analyze cell-type-specific ChIP-Seq data (integrate the Cistrome database) in support of more accurate regulon 
			prediction. DESSO (<a href="https://www.biorxiv.org/content/biorxiv/early/2018/09/16/417378.full.pdf"><b>bioRxiv version</b></a>) is currently 
			under review in Genome Research </li>
			<br>
			</ul>
		</div>
		<div class="clear"></div>
	</div>	

	<div class="qa">
		<h3 class="question"><a id="Q17" style="color:black;">Brower compatibility of IRIS3.</a></h3>
		<div class="answer">
			<img src="./assets/img/tutorial/4.Q16.png">
			<p>IRIS3 is fully functioned on all tested browsers.</p>
			<br>
			<br>
			<br>
			<br>
			<br>
			<br>
			<br>
			<br>			
		</div>
		<div class="clear"></div>
	</div>		
</div></body></html>